File: src/main/java/com/deneb/client/utils/WorldTimer.java
package com.deneb.client.utils;

public class WorldTimer {
    private float OverrideSpeed = 1.0f;

    private void useTimer() {
        if (OverrideSpeed != 1.0f && OverrideSpeed > 0.1f) {
            Wrapper.mc.timer.tickLength = 50.0f / OverrideSpeed;
        }
    }

    public void SetOverrideSpeed(float f) {
        OverrideSpeed = f;
        useTimer();
    }

    public void resetTime() {
        Wrapper.mc.timer.tickLength = 50;
    }
}

File: src/main/java/club/deneb/client/utils/WorldTimer.kt
package club.deneb.client.utils

class WorldTimer {
    private var overrideSpeed = 1.0f
    private fun useTimer() {
        if (overrideSpeed != 1.0f && overrideSpeed > 0.1f) {
            Wrapper.minecraft.timer.tickLength = 50.0f / overrideSpeed
        }
    }

    fun setOverrideSpeed(f: Float) {
        overrideSpeed = f
        useTimer()
    }

    fun resetTime() {
        Wrapper.minecraft.timer.tickLength = 50f
    }
}